### ISSUE TITLE
sublimeLinter-contrib-htmlhint not trigger

### ISSUE THREAD

**Issue by fallenleavesguy** [mmaday/SublimeLinter-contrib-htmlhint#4]
I have installed sublime linter,and htmlhint is in my path.jshint linter,csslint linter work well,but sublimeLinter-contrib-htmlhint doesn't work.I enabled debug mode,there aren't anything about htmlhint,but many information about jshint etc.How should I do?Thank you.

**Comment by mmaday** [comment]
Can you enable debug mode on SublimeLinter (Tools/SublimeLinter/Debug Mode), do View/Show Console, then restart Sublime and send me the console contents?  That should help figure out what's going on.

**Comment by fallenleavesguy** [comment]
ok.The console contents are the same as run htmlhint [files] in shell(mac),It log error about my html file.Is it work right?But debug console log nothing in cmd(windows7).

**Comment by mmaday** [comment]
So I'm not sure I understand.  It's working on your Mac and not Windows 7?  I'd like to see the SublimeLinter log entries when it starts up. There should be entries where it queries for the htmlhint version, etc.

**Comment by yeluoqiuzhi** [comment]
sorry,I want to copy the console contents for you today.But It's working now,I don't know why.I can't find the console contents of yesterday.Thank you!

**Comment by fallenleavesguy** [comment]
I think htmlhint of sublime plugin can work as jshint,It has a gutter,when error is happening,so I don't save my file,and the console log nothing!

**Comment by glen-anderson** [comment]
I'm not sure if this is related but FWIW the plugin stopped working for me when I moved to htmlhint 0.9.8 or newer. Looking at the changelog I can see that they've completely reformated the error/warning text which makes the regex no longer match anything. Moving back to htmlhint 0.9.7 fixes everything.

Debug output with htmlhint 0.9.7:

```
SublimeLinter: htmlhint output:
D:\test.html:
    line 9, col 1: Tag must be paired, Missing: [ </div> ]


1 problems. 
SublimeLinter: match -- msg:"Tag must be paired, Missing: [ </div> ]", match:<_sre.SRE_Match object at 0x0000005FE5B52F08>, line:8, col:0, near:None, warn: None
```

Debug output with htmlhint 0.9.8:

```
SublimeLinter: htmlhint output:

   test.html
      L9 |</body>
          ^ Tag must be paired, missing: [ </div> ], start tag match failed [ <div> ] on line 8. (tag-pair)

1 errors in 1 files 
```

**Comment by mmaday** [comment]
@glen-anderson Thanks for the tip.  That's definitely a problem.  I'll need to update the plugin to handle the latest htmlhint output.

**Comment by chenxuan03** [comment]
SublimeLinter-contrib-htmlhint did work when I installed htmlhint@0.9.7 yesterday. Suddenly, it stopped working because Package Control updated it to the version 1.0.1 .
<img width="946" alt="2015-10-20 2 28 42" src="https://cloud.githubusercontent.com/assets/10443283/10599683/e6563b70-7736-11e5-8d92-d7da5f26b521.png">
htmlhint worked in bash:
<img width="656" alt="2015-10-20 11 36 19" src="https://cloud.githubusercontent.com/assets/10443283/10599698/14b06586-7737-11e5-94b1-bfacae3c2405.png">
htmlhint did not work in Sublime: 
![53b51b5a-9d5f-433a-9f8d-4ff98b6e9ee7](https://cloud.githubusercontent.com/assets/10443283/10599786/ddbd50e2-7737-11e5-815c-53161d2b9f22.png)
console output like this:
<img width="1038" alt="2015-10-20 11 34 25" src="https://cloud.githubusercontent.com/assets/10443283/10599799/f81b883c-7737-11e5-9005-b0a9bea578a1.png">

Please ignore my poor English .  T_T

**Comment by fallenleavesguy** [comment]
I think it only logs in console.Can it work as jshint?I don't know..........

**Comment by chenxuan03** [comment]
@yeluoqiuzhi  htmlhint worked as jshint when sublimeLinter-contrib-htmlhint's version is 1.0.0 yesterday, but I did not take a screenshot.

**Comment by fallenleavesguy** [comment]
I never see that.so,my htmlhint didn't work.How to do?

**Comment by chenxuan03** [comment]
I have the same problem now, so we can wait for @mmaday to give a reply.

**Comment by chenxuan03** [comment]
@yeluoqiuzhi  you can download SublimeLinter-contrib-htmlhint's v1.0.0 zip file , unzip it , rename the folder to "SublimeLinter-contrib-htmlhint" and put it to Packages folder, then you can see htmlhint@0.9.7 working .
<img width="624" alt="2015-10-20 3 15 48" src="https://cloud.githubusercontent.com/assets/10443283/10600670/f2d911b2-773e-11e5-8178-64844a504601.png">
<img width="284" alt="2015-10-20 3 25 20" src="https://cloud.githubusercontent.com/assets/10443283/10600658/db53af02-773e-11e5-8540-d1a4658c0fd4.png">

**Comment by fallenleavesguy** [comment]
It was v1.0.0,but it didn't work

**Comment by mmaday** [comment]
Hi @chenxuan03,

Are you on Windows or Mac?  If possible, could you please upgrade to 1.0.1, htmlhint@0.9.7 and then send me the debug log contents?  To do so, enable debug mode on SublimeLinter (Tools/SublimeLinter/Debug Mode), do View/Show Console, restart Sublime and then please send me the console.

The only thing I updated in 1.0.1 was the version requirement so that htmlhint 0.9.0 - 0.9.7 are the only supported versions.  

In the meantime I'll be updating the plugin to work with the latest htmlhint output format.

**Comment by fallenleavesguy** [comment]
I uninstall htmlhint@0.9.12,then install htmlhint@0.9.7.well done.@mmaday thank you

**Comment by mmaday** [comment]
Hi all,

I've just revamped and released 1.1.0 that works with htmlhint 0.9.8 and greater.  It should hit the package release channel soon.  Please be sure to upgrade to the latest htmlhint once you get the package update by running:
npm install -g htmlhint@latest

If you have any problems, please create a new issue.  Thanks!
